### DISCUSSION TITLE
How to request information about project roles in the Frontend to API scenario

### DISCUSSION THREAD

**Discussion by nikzen** [zitadel/zitadel#7290] [Q&A]
Heyhey👋,

we have the following scenario, it is based on the [concept in your docs](https://zitadel.com/docs/guides/solution-scenarios/frontend-calling-backend-API):

- Our App is a native client in a project of Common Organization
- Our Backend API is a client in a project of a Customer Organization
- The Users are members in the Customer Organization
- The Users are authorized with a Role in the Backend API Project
- The User will use the App to request an Access & ID Token from Zitadel
- The App will send the Zitadel Access Token to the Backend 
- The Backend will do Introspection to check if the User is authorized for the project with the roles etc
- The Backend will then return its own token
- The App will use the backend token to use the Backend

This flow works great, however we want to hide some features in the app, based on the role of the user in the backend project. To do so, the app needs to know the roles, however it seems not possible for the app to request those roles as the app is not part of the project or organization.

Do you have any idea how to solve this? Thanks a lot

**Comment by hifabienne** [comment]
Hei 
I think what you are looking for is the roles scope in combination with the needed project id scope.
You can find a description in the reserved scopes section: https://zitadel.com/docs/apis/openidoauth/scopes#reserved-scopes

Role Scope: urn:zitadel:iam:org:projects:roles
urn:zitadel:iam:org:project:{projectid}:roles

**Reply by nikzen** [reply]
Thanks for the quick reply:

- The Common Org is: `233778719167086596`
- The Project in the Common Org is: `237786901027749896`
- The Customer Org is: `235784326891962376`
- The Project in the Customer Org is: `245784975167717380`

These are the scopes:
- `openid`
- `profile`
- `email`
- `urn:zitadel:iam:org:id:235784326891962376`
- `urn:zitadel:iam:org:project:245784975167717380:roles`
- `urn:zitadel:iam:org:projects:roles`

However the ID Token does not contain the role for the project `245784975167717380` :(

**Reply by hifabienne** [reply]
@livio-a @muhlemmer can you help here?

**Reply by muhlemmer** [reply]
The mentioned scopes, where did you get them from? ID Token or introspection on the API side? Just trying to figure out where you are missing the scope.

Are you calling User Info endpoint from the front end, or are you getting the info directly from the ID Token? In the last case you need to enable the following settings to the front-end app:

![image](https://github.com/zitadel/zitadel/assets/5411563/8fc4c30e-3c29-4fff-b878-651545c404aa)

**Reply by hifabienne** [reply]
The scopes are the ones I sent, and I got them from the reserved scopes documentation

**Reply by nikzen** [reply]
I tried multiple ways, however it does not work. To workaround the issue, we will add `urn:zitadel:iam:org:project:id:zitadel:aud` to the scope and then perform this call https://zitadel.com/docs/apis/resources/auth/auth-service-list-my-user-grants.

From here we can then check if the role is correct. :)
